packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js

```diff
@@ -0,0 +1,299 @@
+import {LightningElement, api, track, wire} from 'lwc';
+import caseClassificationTitle from '@salesforce/label/c.quantic_CaseClassificationTitle';
+import moreTopics from '@salesforce/label/c.quantic_MoreTopics';
+import selectOption from '@salesforce/label/c.quantic_SelectOption';
+import {
+  getObjectInfo,
+  getPicklistValuesByRecordType,
+} from 'lightning/uiObjectInfoApi';
+import CASE_OBJECT from '@salesforce/schema/Case';
+import {
+  registerComponentForInit,
+  initializeWithHeadless,
+} from 'c/quanticHeadlessLoader';
+
+/** @typedef {import("coveo").CaseAssistEngine} CaseAssistEngine */
+/** @typedef {import("coveo").CaseField} CaseField */
+
+/**
+ * The `QuanticCaseClassification` component displays field value suggestion returned by Coveo Case Assist as well as a single-select dropdown containing other available values.
+ *
+ * @category Case Assist
+ * @example
+ * <c-quantic-case-classification engine-id={engineId} field-name="Priority" required label="Which topic is related to your issue?" select-placeholder="More topics" max-choices="4" message-when-value-missing="Select an option"></c-quantic-case-classification>
+ */
+export default class QuanticCaseClassification extends LightningElement {
+  @wire(getObjectInfo, {objectApiName: CASE_OBJECT})
+  objectInfo;
+
+  @wire(getPicklistValuesByRecordType, {
```

Comment:
This gets ALL values for ALL picklist fields?

Is there a way we only get the picklist values for only this specific field?

`fieldName`

Reply:
```js
@api(getPicklistValues, { recordTypeId: '$objectInfo.data.defaultRecordTypeId', fieldApiName: CASE_OBJECT[fieldName]})
    picklistValues;
```

I am really not sure if this works, but maybe it does? 😂

Reply:
It's always been a bit fuzzy the order of execution of things if you modify a variable used in an @wire if the wire refreshes itself correctly

Reply:
Yes exactly, this will get the values  for ALL picklist fields. There is a way to to get only the picklist values for only one specific field but following the doc to achieve this import the field from Salesforce is needed, an import like the following:
`import PRIORITY_FIELD from '@salesforce/schema/Case.Priority';`
and the getting the piclist values like the following:
```
@wire(getPicklistValues, { recordTypeId: RECORD_ID, fieldApiName: PRIORITY_FIELD })
    propertyOrFunction;
```

But since in our case the field can be anything I can't do such import since I don't know the field that the user will want to classify, so I found this workaround of getting all the pickilist values and then using only the values of the field specified by the user.

Reply:
`import PRIORITY_FIELD from '@salesforce/schema/Case.Priority';`
Inspect what the import gives you at runtime, I *think* it might just be the API name of the field.
It might just be that `PRIORITY_FIELD` is `Priority` as a string :P 

Long time since I checked.